mautic/mautic
# [UI/UX] Better table header sorting

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This PR makes sortable headers more visible and adds all necessary attributes for accessibility.

This way:
- When nothing is sorted yet, the button label will tell screen readers "Click to sort rows by [Column name] in ascending order" as the first click always triggers the ascending order
- The accessibility label changes will also trigger changes to the aria-sort attribute
- When you already clicked once, the accessible description becomes "Click to sort rows by [Column name] in descending order"

It adapts to each column name

For users seeing the screen, each sorting cell will show icons (that are hidden for screen readers)


https://github.com/user-attachments/assets/cfcdc633-6060-41c5-9443-7ca3b28c6c05


---
Before:
![image](https://github.com/user-attachments/assets/90a59d58-5835-4cf6-870b-e02d0d0a8af4)







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Contacts
3. Create some
4. Back to the listing page (table)
5. Click on the headers to sort like I did in the video
